cli/cmd/root.go

```diff
@@ -314,13 +318,39 @@ func (options *proxyConfigOptions) validate() error {
 
 	if options.proxyCPURequest != "" {
 		if _, err := k8sResource.ParseQuantity(options.proxyCPURequest); err != nil {
-			return fmt.Errorf("Invalid cpu request '%s' for --proxy-cpu flag", options.proxyCPURequest)
+			return fmt.Errorf("Invalid cpu request '%s' for --proxy-cpu-request flag", options.proxyCPURequest)
 		}
 	}
 
 	if options.proxyMemoryRequest != "" {
 		if _, err := k8sResource.ParseQuantity(options.proxyMemoryRequest); err != nil {
-			return fmt.Errorf("Invalid memory request '%s' for --proxy-memory flag", options.proxyMemoryRequest)
+			return fmt.Errorf("Invalid memory request '%s' for --proxy-memory-request flag", options.proxyMemoryRequest)
+		}
+	}
+
+	if options.proxyCPULimit != "" {
+		cpuLimit, err := k8sResource.ParseQuantity(options.proxyCPULimit)
+		if err != nil {
+			return fmt.Errorf("Invalid cpu limit '%s' for --proxy-cpu-limit flag", options.proxyCPULimit)
+		}
+		if options.proxyCPURequest != "" {
+			// Not checking for error because option proxyCPURequest was already validated
+			if cpuRequest, _ := k8sResource.ParseQuantity(options.proxyCPURequest); cpuRequest.MilliValue() > cpuLimit.MilliValue() {
+				return fmt.Errorf("The cpu limit '%s' cannot be lower than the cpu request '%s'", options.proxyCPULimit, options.proxyCPURequest)
+			}
+		}
+	}
+
+	if options.proxyMemoryLimit != "" {
+		memoryLimit, err := k8sResource.ParseQuantity(options.proxyMemoryLimit)
+		if err != nil {
+			return fmt.Errorf("Invalid memory limit '%s' for --proxy-memory-limit flag", options.proxyMemoryLimit)
+		}
+		if options.proxyMemoryRequest != "" {
+			// Not checking for error because option proxyMemoryRequest was already validated
+			if memoryRequest, _ := k8sResource.ParseQuantity(options.proxyMemoryRequest); memoryRequest.MilliValue() > memoryLimit.MilliValue() {
```

Comment:
Sorry I wasn't clearer on my previous comment. CPU values are fractions of 1 (like 100m means 10% of one CPU core) so we should use `MilliValue`. Memory on the other hand is an integer, in bytes. So we can still use `Value` for memory, while saving us some cycles.

Reply:
Though I have no problem making that change, isn't it better to opt for consistency for something that would save an insignificant amount of cycles?

Regardless, fixed with 35a4d13